### ISSUE TITLE
An option to change error message to error stack

### ISSUE THREAD

**Issue by sompylasar** [SamVerschueren/listr#74]
Currently only the error message is available:
https://github.com/SamVerschueren/listr/blob/88dca6293690b560509fdcfe253b93a6c32ae01c/lib/task.js#L184-L192

and in case a dumb JavaScript error like `Cannot convert undefined or null to object` leaks through, this message is the only thing available to debug.

With the [`listr-multiline-renderer`](https://github.com/entwicklerstube/listr-multiline-renderer), the multiline stack is properly printed.

**Comment by SamVerschueren** [comment]
The error is [thrown further upstream](https://github.com/SamVerschueren/listr/blob/master/lib/task.js#L198), so you are able to catch it and print it after the tasks exit.

```js
'use strict';
const Listr = require('listr');

const tasks = new Listr([
	{
		title: 'Do something',
    task: () => {
      throw new Error('foo bar');
    }
	}
]);

tasks.run().catch(err => {
	console.error(err);
});
```

Which results in:
<img width="904" alt="screen shot 2017-09-11 at 12 10 20" src="https://user-images.githubusercontent.com/1913805/30269733-34ae5800-96ea-11e7-971f-7e5e582eaeff.png">

**Comment by sompylasar** [comment]
But this doesn't seem to apply if I'm setting exitOnError to false (which I need to do).

**Comment by SamVerschueren** [comment]
So just to be clear, you want to be able to show the error stack in the `listr-update-renderer`?

Currently, only the [error message is sent](https://github.com/SamVerschueren/listr/blob/master/lib/task.js#L189-L192) to the renderers. What I could do though is emitting the entire error.

```js
this.next({
	type: 'ERROR',
	data: err
});
```

That way, you have more control in a custom renderer as I don't think I will ever print the entire stack trace in the `update-renderer`. It will totally break the UI.

**Comment by sompylasar** [comment]
Yes, in the [`listr-multiline-renderer`](https://github.com/entwicklerstube/listr-multiline-renderer) which seems to be a fork of `listr-update-renderer` that allows multiline output.

Please let the users decide whether the stack breaks the UI or not. You can have an option for that (e.g. a `renderErrorStack` or `errorRenderMode` option) but not just silently discard useful bits of information that can help find the root cause of an error.

**Comment by SamVerschueren** [comment]
I'm 👍 with this. Requires updates in `listr-update-renderer` and in `listr` itself to dispatch the full errors.

**Comment by moehlone** [comment]
Any updates on this? Would appreciate it too, if the error is passed completely.

**Comment by axe312ger** [comment]
I'd love to see this, too :)
